webapp/src/main/java/io/github/microcks/util/dispatcher/ProxySpecification.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright The Microcks Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.github.microcks.util.dispatcher;
+
+import com.fasterxml.jackson.annotation.JsonPropertyOrder;
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+/**
+ * Represent the specification of a Proxy evaluation. <code>dispatcher</code> should be the original dispatcher to apply
+ * with its <code>dispatcherRules</code> companion. If no response actually found by mock controller, the original
+ * request will be forwarded to the <code>proxyUrl</code>.
+ */
+@JsonPropertyOrder({ "dispatcher", "dispatcherRules", "proxyUrl" })
```

Comment:
I'm not sure why this class is needed as, basically, I think the dispatcher specification needs a backend or proxy URL that could fit into the `dispatcherRules` property living at the operation level. We introduced this complex structure for `FALLBACK` as it acts like a try-catch:
* 1st try the original dispatcher specified in `dispatcher` and using `dispatcherRules`
* 2nd if 1st is not found, we finally use the `fallback` to get the default response name.

So that for fallback, we ended up with:

```yaml
  /pastry/{name}:
    summary: Global operations on pastries
    get:
      x-microcks-operation:
        dispatcher: FALLBACK
        dispatcherRules: |
          {
            "dispatcher": "URI_PARTS",
            "dispatcherRules": "name",
            "fallback": "Our default (not found - 404) response"
          }
```

I don't think this complexity is needed for proxy stuff but I may miss something.

Reply:
Please check out my comment below